### PR TITLE
models/core app

### PR DESCRIPTION
**Pull request by Walrussuit101**
Not 100% done with this branch yet, but I wanted to get it merged into develop because of the addition of the "core" app. Which is vital for development.

Added wiki sys/personal page models (only fields/relations). All you should need to do to update your db is cli into django container and run './manage.py migrate' (god bless migration files)

### CODE REVIEW

**Comment by Rithonn** [comment]
If the branch is not 100% complete then we should wait to merge it. I understand that there is a core item that should be in develop, but we should wait until the branch has been completed in its entirety before merging. For the future if there is a PR that is implementing two items such as this and one should be merged before the other we should work on splitting the two into seperate tickets so not to have to wait on one thing or the other.

**Comment by Rithonn** [comment]
We will also need to create tickets for anything that is done in this PR if there is none that exist currently for them. Make sure to assign yourself to them and update them as progress flows.

**Comment by Walrussuit101** [comment]
100% done. 
- added other models from diagram.
- made separate issues for model creation/core app creation.

File: swarmdjango/core/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                ('content', django.contrib.postgres.fields.jsonb.JSONField()),
+            ],
+        ),
+        migrations.CreateModel(
+            name='User',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('username', models.TextField()),
+                ('password', models.TextField()),
+                ('email', models.TextField()),
```
Comment (Rithonn) [inline swarmdjango/core/migrations/0001_initial.py:73]: Do we want to have TextField here or should we use EmailField? https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.EmailField

File: swarmdjango/core/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        migrations.CreateModel(
+            name='Wiki',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('title', models.TextField()),
+                ('briefDescription', models.TextField()),
+                ('entries', models.ManyToManyField(to='core.Entry')),
+            ],
+        ),
+        migrations.CreateModel(
```
Comment (Rithonn) [inline swarmdjango/core/migrations/0001_initial.py:105]: Do we want to have the file upload field added to this or should we worry about file storage fields later? https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.FileField

Comment (Rithonn) [reply swarmdjango/core/migrations/0001_initial.py:105]: We could also use the built in image field as well https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ImageField

File: swarmdjango/core/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            ],
+            bases=('core.user',),
+        ),
+        migrations.CreateModel(
+            name='SponsorPersonalPage',
+            fields=[
+                ('personalpage_ptr', models.OneToOneField(auto_created=True, on_delete=django.db.models.deletion.CASCADE, parent_link=True, primary_key=True, serialize=False, to='core.PersonalPage')),
+                ('missionStatement', models.TextField()),
+                ('reasonForSponsorship', models.TextField()),
+                ('companyLink', models.TextField()),
```
Comment (Rithonn) [inline swarmdjango/core/migrations/0001_initial.py:92]: https://docs.djangoproject.com/en/3.1/ref/models/fields/#urlfield

File: swarmdjango/core/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            ],
+            bases=('core.personalpage',),
+        ),
+        migrations.CreateModel(
+            name='Wiki',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
```
Comment (Rithonn) [inline swarmdjango/core/migrations/0001_initial.py:99]: UUID field might be better in this instance with a completely unique id here. https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.UUIDField

File: swarmdjango/core/migrations/0001_initial.py (hunk 2 above)
Comment (Walrussuit101) [reply swarmdjango/core/migrations/0001_initial.py:105]: Should probably use ImageField, but I think we'll have to look into that later.
We'll need to know where we're storing it/how (mainly for the 'ImageField.upload_to' option)

**Pull request merged**